lib/routes/sms.js

```diff
@@ -76,15 +76,17 @@ module.exports = (log, isA, error, config, customs, sms) => {
         function sendMessage (senderId) {
           return sms.send(phoneNumber, senderId, messageId, acceptLanguage)
             .catch(err => {
-              if (err.status === 500) {
-                throw error.messageRejected(err.reason, err.reasonCode)
+              if (err) {
+                if (err.status === 500) {
+                  throw error.messageRejected(err.reason, err.reasonCode)
+                }
+
+                if (err.status === 400) {
+                  throw error.invalidMessageId()
+                }
               }
 
-              if (err.status === 400) {
-                throw error.invalidMessageId()
-              }
-
-              throw new error.unexpectedError()
```

Comment:
This shouldn't actually cause an error, thanks to JavaScript!

The `new` keyword will create a new `[object Object]`, give it the prototype of `error.unexpectedError.prototype`, and then pass it like `error.unexpectedError.call(theNewObj)`. And then, since that function doesn't do anything with `this`, and instead returns a totally different value, the new `[object Object]` just gets dumped on the floor. Unloved.

(Played in console just to be sure.)